### PR TITLE
oric1_cass.xml: Software list additions

### PR DESCRIPTION
**Pull request by ArcadeShadow**
New working software list additions
--------------------------------------------
4K Kong [The Oric Site]
Cobra [The Oric Site]
Impossible Mission (Atmos 50Hz) [Defence Force]
Impossible Mission (Atmos 60Hz) [Defence Force]
Invaders (Program Factory) [The Oric Site]
Le Fer d'Amnukor [The Oric Site]
Oricium [Defence Force]
Rush Hour 4K [The Oric Site]
Skool Daze [The Oric Site]
Stormlord [Defence Force]
Stormlord (alt) [The Oric Site]
The Pits [The Oric Site]
Transat One [The Oric Site]

New NOT working software list additions
--------------------------------------------
Arsène Lapin [The Oric Site]
Detective Story [The Oric Site]
Composer [The Oric Site]

Redump software
--------------------------------------------
Le Secret du Tombeau [The Oric Site]
Le Tour du Monde en 80 Jours [The Oric Site]
Mission Impossible [The Oric Site]
The Ultra [The Oric Site]
Contract Bridge [The Oric Site]

Removed software (merge as B-side of the sets)
--------------------------------------------
Contract Bridge (doc)
Fire Flash (alt)
Le Secret du Tombeau (alt)
Le Tour du Monde en 80 Jours (demo)
Mission Impossible (alt)
The Ultra (alt)

### CODE REVIEW

File: hash/oric1_cass.xml [hunk 1]
```diff
@@ -86,9 +86,23 @@ license:CC0-1.0
...
+			<dataarea name="cass" size="4093">
+				<rom name="4KKong.tap" size="4093" crc="1335cca5" sha1="fc52ce11b70ab3a95d7780b33e6ed95ae3aea150"/>
+			</dataarea>
+		</part>
+	</software>
+
 	<software name="3dtunnel">
 		<description>The 3D Tunnel</description>
-		<year>1998</year>
+		<year>1998</year>
```
Comment (cuavas) [inline hash/oric1_cass.xml:105]: Stray whitespace at the end of this line.

File: hash/oric1_cass.xml [hunk 2]
```diff
@@ -262,10 +276,34 @@ license:CC0-1.0
 		</part>
 	</software>
 
+	<software name="arsenelp" supported="no">
+		<description>Arsène Lapin</description>
+		<year>1984</year>
+		<publisher>Infogrames</publisher>
+		<notes><![CDATA[
+Cassette with two versions, side 1 with the Atmos version, side 2 with Oric-1 version.
+]]></notes>
```
Comment (cuavas) [inline hash/oric1_cass.xml:285]: Isn’t that apparent from the part IDs anyway?

File: hash/oric1_cass.xml [hunk 3]
```diff
@@ -262,10 +276,34 @@ license:CC0-1.0
...
+			</dataarea>
+		</part>
+	</software>
+
 	<software name="asciichase">
 		<description>ASCII Chase</description>
 		<year>2017</year>
-		<publisher>Fabrizio Caruso</publisher>
+		<publisher>&lt;homebrew&gt;</publisher>
+		<info name="developer" value="Fabrizio Caruso" />
```
Comment (cuavas) [inline hash/oric1_cass.xml:306]: I don’t understand the motivation for treating “homebrew” as a publisher.  If the developer distributes the software online themselves, isn’t it effectively self-published?  Can someone give a good reason for the fake “homebrew” publisher?

File: hash/oric1_cass.xml [hunk 4]
```diff
@@ -742,6 +780,29 @@ Type-in from magazine
 		</part>
 	</software>
 
+	<software name="cobra">
+		<description>Cobra</description>
+		<year>1985</year>
+		<publisher>Norsoft</publisher>
+		<notes><![CDATA[
+Cassette with two versions, side 1 with the Atmos version, side 2 with Oric-1 version.
+]]></notes>
```
Comment (cuavas) [inline hash/oric1_cass.xml:789]: Once again, isn’t this apparent from the part IDs?

File: hash/oric1_cass.xml [hunk 5]
```diff
@@ -1849,6 +2006,30 @@ Type-in from magazine
...
+		<description>Impossible Mission (Atmos 60Hz)</description>
+		<year>2010</year>
+		<publisher>Defence Force</publisher>
+		<info name="usage" value="only Oric Atmos compatible" />
+		<part name="cass" interface="oric1_cass">
+			<dataarea name="cass" size="48110">
+				<rom name="im10-60.tap" size="48110" crc="225c6d82" sha1="09009d75d67e7e7501561028898559a68280820e"/>
+			</dataarea>
+		</part>
+	</software>
```
Comment (cuavas) [inline hash/oric1_cass.xml:2031]: Wouldn’t compatibility be handled better with compatibility tags?

File: hash/oric1_cass.xml [hunk 6]
```diff
@@ -2439,24 +2631,21 @@ The cassette stops during the game loading
...
-	<software name="missimp1" cloneof="missimp">
-		<description>Mission Impossible (alt)</description>
-		<year>1984</year>
-		<publisher>No Man's Land</publisher>
-		<part name="cass" interface="oric1_cass">
-			<dataarea name="cass" size="39135">
-				<rom name="Mission Impossible (1984)(No Man's Land)(FR)[a no title screen].tap" size="39135" crc="b66e6ab6" sha1="f879ee133a822a3cb23b5eeccb1cd8b7fa37551e"/>
+			<feature name="part_id" value="FACE A - ORIC 48K"/>
+			<dataarea name="cass" size="47175">
+				<rom name="Mission Impossible - Face A.tap" size="47175" crc="1aba6c78" sha1="8b8ac2594946a2fdbfe30520b392a6b335f9ec60"/>
```
Comment (cuavas) [inline hash/oric1_cass.xml:2648]: Is this missing one side?  Should it have a placeholder for the other side?

File: hash/oric1_cass.xml [hunk 7]
```diff
@@ -4068,20 +4341,21 @@ The cassette stops during the game loading
 		<description>The Ultra</description>
 		<year>1983</year>
 		<publisher>Personal Software Services</publisher>
-		<part name="cass" interface="oric1_cass">
-			<dataarea name="cass" size="9078">
-				<rom name="Ultra, The (1983)(Personal Software Services)(GB)[no title screen].tap" size="9078" crc="a45726b7" sha1="fcc1cce201d9a9ee51a4446494cd80431068b425"/>
+		<notes><![CDATA[
+Cassette with two versions, side 1 with the Oric 16/48K version, side 2 with Oric V1.1 & Atmos version.
+]]></notes>
```
Comment (cuavas) [inline hash/oric1_cass.xml:4346]: Isn’t this apparent from the part IDs?

**Review by cuavas — COMMENTED** [review]
Please also give the rationale for removing the items that were removed.

File: hash/oric1_cass.xml (hunk 3 above)
Comment (ArcadeShadow) [reply hash/oric1_cass.xml]: If that's the rule, I made the changes.
I thought that if it was developed and self-published, the term "homebrew" would be appropriate.
(BTW, I think there are more software lists that use “homebrew” as a publisher)

File: hash/oric1_cass.xml (hunk 6 above)
Comment (ArcadeShadow) [reply hash/oric1_cass.xml]: I've added the missing B-side as "nodump" with size=0 (I think that's the way it is done)

File: hash/oric1_cass.xml (hunk 5 above)
Comment (ArcadeShadow) [reply hash/oric1_cass.xml]: You're right!

**Comment by ArcadeShadow** [comment]
> ive the rationale for removing the items that were remove

Those items are the B sides of the cassette. 
So I think they would be better on part2 of the set, instead of separating them into two sets.
For that purpose, I used the The Oric Site dump.
That was the reason why I done that.

File: hash/oric1_cass.xml (hunk 3 above)
Comment (cuavas) [reply hash/oric1_cass.xml]: I know there are others.  I can understand the reason for using it for a system that wasn’t designed to allow self-hosted development, or was only designed to run licensed software.  For example the NES and Game Boy were intended to only run licensed software and had no provisions for users to develop their own software.  Homebrew software for these systems required the use of reverse-engineered development kits, Flash cartridges, etc. so it’s a completely different category to licensed cartridges, or even unlicensed and bootleg cartridges.

There’s also a distinction with the PlayStation, where homebrew software developed with the Net Yaroze system was far more restricted than licensed software.

But for an actual computer system, there’s effectively no difference.  For 8-bit home computers in particular, a lot of commercial software was developed by individuals, using the same tools available to everyone.  The distinction is completely artificial.

**Pull request merged**